### PR TITLE
Extract ProfilerInterface from ProfilerBase, and put more logic in the latter

### PR DESCRIPTION
**Pull request by Jongy**
## Description
Merge reusable logic into `ProfilerBase`.

## Motivation and Context
Improving the codebase.

## How Has This Been Tested?
CI

## Checklist:
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have updated the relevant documentation.
- [ ] I have added tests for new logic.

### CODE REVIEW

File: gprofiler/profiler_base.py [hunk 1]
```diff
@@ -31,10 +38,40 @@ def __exit__(self, exc_type, exc_val, exc_tb):
         self.stop()
 
 
-class NoopProfiler(ProfilerBase):
+class ProfilerBase(ProfilerInterface):
```
Comment (michelhe) [inline gprofiler/profiler_base.py:41]: ABC?

Comment (Jongy) [reply gprofiler/profiler_base.py:41]: We don't need anything that `ABCMeta` provides

**Comment by Jongy** [comment]
> Rebase this on #106 ?

If we merge it first :)

**Comment by Jongy** [comment]
Meh I'll just rebase, yes

**Comment by michelhe** [comment]
LGTM

**Review by michelhe — APPROVED** [review]

**Pull request merged**